src/main/java/pl/akai/bookcrossing/list/BookRestController.java

```diff
@@ -32,20 +31,27 @@ public String booksList(Model model) {
 
     @GetMapping("/book/add")
     public String addBookForm(Model model) {
+        String newTags = "";
+        String existingTags = "";
+        model.addAttribute("tag", newTags);
+        model.addAttribute("extag", existingTags);
         model.addAttribute("book", new Book());
+        model.addAttribute("tags", bookBean.getAllTags());
         return "form";
     }
 
     @PostMapping("/book/add")
-    public String addBookSubmit(@ModelAttribute Book book, Model model) {
-        model.addAttribute("book", book);
+    public String addBookSubmit(@RequestParam(name = "tag", required = false, defaultValue = "") String newTags, @RequestParam(name = "extag", required = false, defaultValue = "") String existingTags, @ModelAttribute Book book, Model model) {
```

Comment:
To do najpiękniejszych rzeczy nie należy. Lepszym pomysłem byłby nowy model,który by rozszerzał Book np. BookFormResponese, który miałby dodatkowe pole extag (nazwa do zmiany, bo jest mało mówiąca) będące List<Integer> oraz tag (nazwa również do zmiany bo nie wsazuje że to nowe tagi) będące Stringiem.